code/modules/mob/living/silicon/robot/component.dm

```diff
@@ -34,34 +36,62 @@
 	installed = -1
 	uninstall()
 
-/datum/robot_component/proc/take_damage(brute, electronics, sharp)
+/datum/robot_component/proc/take_damage(brute, electronics, sharp, updating_health = TRUE)
```

Comment:
I really hate the robot components. They're a datum that is related to an item; the robots hold the datum, then generate the item.

Not asking you to do this, but it would make more sense for all these behaviors to be handled on the item, then have a list of `robot_components` that hold the items. Having it wrapped like this is just weird.

Reply:
That's a refactor for another PR